hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/HandleContextImpl.java

```diff
@@ -114,8 +136,10 @@ public HandleContextImpl(
             @NonNull final TransactionDispatcher dispatcher,
             @NonNull final ServiceScopeLookup serviceScopeLookup,
             @NonNull final BlockRecordInfo blockRecordInfo,
-            @NonNull final RecordCache recordCache) {
-        this.txBody = requireNonNull(txBody, "txBody must not be null");
+            @NonNull final RecordCache recordCache,
+            @NonNull final FeeManager feeManager,
+            @NonNull final Instant userTransactionConsensusTime) {
```

Comment:
The `recordBuilder` already contains the `userTransactionConsensusTime`.

Reply:
I'm not sure if I like that better or not, but I do hate that we have a billion arguments to this constructor. Something to look at in the future to clean this up.